### PR TITLE
Catalog to uv search

### PR DESCRIPTION
**Pull request by kirkkwang**
ref: #183

iiif_print_183

[Add a highlight detection](https://github.com/scientist-softserv/iiif_print/commit/172e13014679ef6f8806e58a69922f96192da49f) 

The `#has_any_highlighting?` method is used to determine if the search
returned any highlighting. If it did, then we add the parent_query to
the search params so that the search results.  This will avoid causing
and No matches found notification in the UV search.  Also in this
commit, we change `hl.fl` to return all fields that are indexed.

---

[Add IIIF Helpers into the load path](https://github.com/scientist-softserv/iiif_print/commit/1250144d026fd58ede24c60456a2808ee25dce13) 

This commit makes the IIIF Helpers work by loading in the engine.rb.
The main reason for the change is the get the #uv_search_param method
usable for the Universal Viewer.  This parameter will automatically get
the user into the search term they were using for the catalog search.

---

[Add catalog to uv search instructions to README](https://github.com/scientist-softserv/iiif_print/commit/0b5b61224a58d7859b57b4e8653e09cedd388e46)

---

[Add configuration for uv config and uv base path](https://github.com/scientist-softserv/iiif_print/commit/575c5e5e330e86ede8f6b188d99c962f70cd8f66) 

This commit will add configuration for the Universal Viewer config and
base path. This solve the problem where some applications use
`uv-config.json` while other maybe use `uv_config.json`.  While we're in
there, we also add a configuration for the Universal Viewer base path.

---

[Add a way to suppress No matches were found](https://github.com/scientist-softserv/iiif_print/commit/78fc6fc787907192be09bd17681f9eaebf642a2e) 

This commit will allow the search to bail if no matches were found
instead of going through an returning no results.

### CODE REVIEW

File: app/models/iiif_print/iiif_search_response_decorator.rb [hunk 1]
```diff
@@ -4,7 +4,11 @@ module IiifSearchResponseDecorator
     # @see https://github.com/scientist-softserv/louisville-hyku/commit/67467e5cf9fdb755f54419f17d3c24c87032d0af
     def annotation_list
       json_results = super
-      json_results&.[]('resources')&.each do |result_hit|
+
```
Comment (ShanaLMoore) [inline app/models/iiif_print/iiif_search_response_decorator.rb:7]: https://github.com/boston-library/blacklight_iiif_search/blob/master/spec/models/blacklight_iiif_search/iiif_search_response_spec.rb

**Review by ShanaLMoore — APPROVED** [review]

File: app/models/concerns/iiif_print/solr/document.rb [hunk 2]
```diff
@@ -49,4 +49,8 @@ def respond_to_missing?(method_name, include_private = false)
   def file_set_ids
     self['file_set_ids_ssim']
   end
+
```
Comment (ShanaLMoore) [inline app/models/concerns/iiif_print/solr/document.rb:52]: https://github.com/samvera/hyrax/blob/2b807fe101176d594129ef8a8fe466d3d03a372b/spec/models/solr_document_spec.rb

File: app/helpers/iiif_print/iiif_helper_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
```
Comment (ShanaLMoore) [inline app/helpers/iiif_print/iiif_helper_decorator.rb:1]: https://github.com/scientist-softserv/iiif_print/blob/main/spec/helpers/hyrax/iiif_helper_spec.rb

File: app/search_builders/concerns/iiif_print/highlight_search_params.rb [hunk 4]
```diff
@@ -6,9 +6,10 @@ module HighlightSearchParams
     def highlight_search_params(solr_parameters = {})
       return unless solr_parameters[:q] || solr_parameters[:all_fields]
       solr_parameters[:hl] = true
-      solr_parameters[:'hl.fl'] = 'all_text_tsimv'
+      solr_parameters[:'hl.fl'] = '*'
```
Comment (jeremyf) [inline app/search_builders/concerns/iiif_print/highlight_search_params.rb:9]: Non-blocker: talk to me about this.

**Review by jeremyf — APPROVED** [review]

File: app/search_builders/concerns/iiif_print/highlight_search_params.rb (hunk 4 above)
Comment (kirkkwang) [reply app/search_builders/concerns/iiif_print/highlight_search_params.rb:9]: This is suppose to add all indexed fields into hl.fl which has to do with highlighting.  I'm not too sure of how this works to be honest but it was something Rob told us to change.

**Pull request merged**